eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/AbstractMutableBiMap.java

```diff
@@ -597,13 +597,15 @@ public MutableShortBag collectShort(ShortFunction<? super V> shortFunction)
     @Deprecated
     public <S> MutableSet<Pair<V, S>> zip(Iterable<S> that)
     {
-        return this.delegate.zip(that, new UnifiedSet<>());
+        int thatSize = Iterate.sizeOf(that);
+        UnifiedSet<Pair<V, S>> target = UnifiedSet.newSet(Math.min(this.size(), thatSize));
+        return this.delegate.zip(that, target);
     }
 
     @Override
     public MutableSet<V> select(Predicate<? super V> predicate)
     {
-        return this.delegate.select(predicate, new UnifiedSet<>());
+        return this.delegate.select(predicate, UnifiedSet.newSet(this.size()));
```

Comment:
We cannot pre-size for select() as we do not know beforehand for how many elements the Predicate will return true. Can you please revert this?

Reply:
Sorry for this. Reverted.